docs/examples/wasm/webpack/src/secure-cell.js

```diff
@@ -0,0 +1,407 @@
+            adjustTextAreaRows(plaintextArea)
+
+            errorBoxTokenProtect.textContent = ''
+            errorBoxTokenProtect.classList.add('hidden')
+        }
+        catch (error) {
+            plaintextArea.value = ''
+            plaintextLength.textContent = '0 bytes'
+
+            errorBoxTokenProtect.innerHTML = massageErrorMessage(error)
+            errorBoxTokenProtect.classList.remove('hidden')
+        }
+    }
+
+    function encryptContextImprint() {
+        try {
+            const decodedKey = Base64ToBytes(symmetricKeyInput.value)
+            const encodedContext = StringToUTF8Array(contextArea.value)
+            const encodedPlaintext = StringToUTF8Array(plaintextArea.value)
+
+            const cell = themis.SecureCellContextImprint.withKey(decodedKey)
+            const ciphertext = cell.encrypt(encodedPlaintext, encodedContext)
+
+            ciphertextContextImprintArea.value = BytesToBase64(ciphertext)
+            ciphertextContextImprintLength.textContent = `${ciphertext.length} bytes`
+            adjustTextAreaRows(ciphertextContextImprintArea)
+
+            errorBoxContextImprint.textContent = ''
+            errorBoxContextImprint.classList.add('hidden')
+        }
+        catch (error) {
+            ciphertextContextImprintArea.value = ''
+            ciphertextContextImprintLength.textContent = '0 bytes'
+
+            errorBoxContextImprint.innerHTML = massageErrorMessage(error)
+            errorBoxContextImprint.classList.remove('hidden')
+        }
+    }
+
+    function decryptContextImprint() {
+        try {
+            const decodedKey = Base64ToBytes(symmetricKeyInput.value)
+            const encodedContext = StringToUTF8Array(contextArea.value)
+            const decodedCiphertext = Base64ToBytes(ciphertextContextImprintArea.value)
+
+            const cell = themis.SecureCellContextImprint.withKey(decodedKey)
+            const plaintext = cell.decrypt(decodedCiphertext, encodedContext)
+
+            plaintextArea.value = UTF8ArrayToString(plaintext)
+            plaintextLength.textContent = `${plaintext.length} bytes`
+            adjustTextAreaRows(plaintextArea)
+
+            errorBoxContextImprint.textContent = ''
+            errorBoxContextImprint.classList.add('hidden')
+        }
+        catch (error) {
+            plaintextArea.value = ''
+            plaintextLength.textContent = '0 bytes'
+
+            errorBoxContextImprint.innerHTML = massageErrorMessage(error)
+            errorBoxContextImprint.classList.remove('hidden')
+        }
+    }
+
+    function encryptWithKey() {
+        encryptSealWithKey()
+        encryptTokenProtect()
+        encryptContextImprint()
+    }
+
+    function encryptWithPassphrase() {
+        encryptSealPassphrase()
+    }
+
+    let encrypt = encryptWithKey
+    let decryptSeal = decryptSealWithKey
+
+    let update = function(){}
+
+    generateButton.addEventListener('click', function() {
```

Comment:
so pretty code structure and readable variable names... so easy to read and understand, thank you. cool example project